libp2p/builders.nim

```diff
@@ -149,16 +166,19 @@ proc build*(b: SwitchBuilder): Switch =
   if isNil(b.rng):
     b.rng = newRng()
 
-  let switch = newSwitch(
-    peerInfo = peerInfo,
-    transports = transports,
-    identity = identify,
-    muxers = muxers,
-    secureManagers = secureManagerInstances,
-    maxConnections = b.maxConnections,
-    maxIn = b.maxIn,
-    maxOut = b.maxOut,
-    maxConnsPerPeer = b.maxConnsPerPeer)
+  let switch = try:
+    newSwitch(
+      peerInfo = peerInfo,
+      transports = transports,
+      identity = identify,
+      muxers = muxers,
+      secureManagers = secureManagerInstances,
+      maxConnections = b.maxConnections,
+      maxIn = b.maxIn,
+      maxOut = b.maxOut,
+      maxConnsPerPeer = b.maxConnsPerPeer)
+  except CatchableError as exc:
+    raise newException(Defect, exc.msg)
```

Comment:
should this really be a defect? isn't this when the port is busy for example?

Reply:
Good point, in fact I'm propagating `LPError` already, so it should not `Defect` here.